Sources/ReleaseServer/ReleaseServer.WebApi/Repositories/ReleaseArtifactRepository.cs

```diff
@@ -70,8 +69,13 @@ public async Task StoreArtifact(ReleaseArtifactModel artifact)
                 Directory.CreateDirectory(Path.Combine(ArtifactRoot, artifact.ProductInformation.ProductIdentifier));
                 
                 //Move the extracted payload to the right directory
-                Directory.Move(tmpPath, path);
+                Directory.Move(tmpDir.ToString(), path);
                 Logger.LogInformation("The Artifact was successfully stored");
+                
+                //Cleanup the tmp directory 
+                Directory.Delete(tmpDir.Parent.ToString());
```

Comment:
This should be call with `recursive = true` as otherwise it will fail if there still exists content inside this directory by any reason.
In regard to clean code you could just use `DirectoryInfo.Delete()` for this:

```suggestion
                tmpDir.Parent.Delete(true);
```

Reply:
Realized, see:
https://github.com/Traeger-GmbH/release-server/blob/feature/implement-backup-mechanism/Sources/ReleaseServer/ReleaseServer.WebApi/Repositories/ReleaseArtifactRepository.cs#L75-L76